trie.c

```diff
@@ -65,12 +75,12 @@ bool st_find_longest_chain(st_trie_t *trie, st_key_buffer_t *search, st_trie_pay
 	if (code & TRIE_MATCH_BIT) {
         // match nodes are side attachments, so decrease depth
         depth--;
+        // record this as the new longest match
+        longest_match->trie_match_index = offset;
```

Comment:
these two lines need to be moved to after the travel down st_find_longest_chain, since we will then know we're at the longest sequence.

Reply:
Perhaps `longest_match` should be renamed to `longest_match_sofar`. This logic is correct. The last match found is always the longest match, because we traverse the buffer linearly. All recursive calls in this function are tail calls.

Reply:
The logic is correct either way, but we only have to store the indices after visiting lower down the trie and not finding a match there. This is a small optimization which leads to not needlessly overwriting those two values on each match, since we only want the longest match.